### PR TITLE
Bugfix/cldsrv 533

### PR DESCRIPTION
**Pull request by ghost**
Fixes a limitation during a corner case, when restoring an object within account/bucket with enabled exceeded quota:

* First, we reserve space when processing `RestoreObject`
* When the data is being actually restored, the cold backend will PUT the object tot he HOT location (with a specific header, `x-scal-s3-version-id`). But then tries to increase the inflights: in this case, if the quota is exceeded already, or the bytes are near to the limit, we fail, even if the space was reserved.

This PR introduces a way to detect DMF calls (the specific header) and, in this case, only check the quotas, without increasing the in-flights.

### CODE REVIEW

**Review by benzekrimaha — APPROVED** [review]
LGTM

File: lib/metadata/metadataUtils.js [hunk 1]
```diff
@@ -242,13 +242,16 @@ function standardMetadataValidateBucketAndObj(params, actionImplicitDenies, log,
             const needQuotaCheck = requestType => requestType.some(type => actionNeedQuotaCheck[type] ||
                 actionWithDataDeletion[type]);
             const checkQuota = params.checkQuota === undefined ? needQuotaCheck(requestType) : params.checkQuota;
-            if (!checkQuota) {
+            // withVersionId cover cases when an object is being restored with a specific version ID.
+            // In this case, the storage space was already accounted for when the RestoreObject API call
+            // was made, so we don't need to add any inflight, but quota must be evaluated.
+            if (!checkQuota && !withVersionId) {
```
Comment (francoisferrand) [inline lib/metadata/metadataUtils.js:248]: is this really needed?

Comment (francoisferrand) [inline lib/metadata/metadataUtils.js]: is this really needed?


```suggestion
            if (!checkQuota) {
```

Comment (ghost) [reply lib/metadata/metadataUtils.js]: Indeed, it had been removed (and squashed)

**Review by francoisferrand — APPROVED** [review]

**Pull request merged**